src/server.rs

```diff
@@ -461,7 +461,15 @@ impl Server {
             None => &user.username,
         };
 
-        let password = user.server_password.as_ref();
+        #[allow(clippy::match_as_ref)]
```

Comment:
Here comes the clippy overrides! Inevitable I guess, but that's why I avoided clippy for the longest time :)

Reply:
@levkk I agree, it's not pretty, but hopefully we'll have only a few of those.
In my experience, the only clippy override that's hard to avoid is `#[allow(clippy::too_many_arguments)]`.

BTW, created https://github.com/postgresml/pgcat/pull/614 to remove these overrides.